sys/kern/vfs_vnops.c

```diff
@@ -1108,8 +1109,8 @@ vn_io_fault1(struct vnode *vp, struct uio *uio, struct vn_io_fault_args *args,
 		}
 		if (len > io_hold_cnt * PAGE_SIZE)
 			len = io_hold_cnt * PAGE_SIZE;
-		addr = (__cheri_addr vaddr_t)uio_clone->uio_iov->iov_base;
-		end = round_page(addr + len);
+		addr = uio_clone->uio_iov->iov_base;
+		end = roundup2(addr + len, PAGE_SIZE);
```

Comment:
round_page should work just fine?

Reply:
Yeah, this was before I decided I should really just do #719. I'll fix up.